pkg/types/types.go

```diff
@@ -270,7 +271,25 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric, mismatchCheck bool) (metric Metric, mismatches int) {
+func getPointMajorityValue(values []float64) (majorityValue float64, mismatch bool) {
+	valuesCount := len(values)
+	if valuesCount == 0 {
+		return 0, false
+	}
+
+	valuesToCount := make(map[float64]int)
```

Comment:
I have changed the logic so that this function is called when there is an observed mismatch. Additionally, I've set the initial bucket size of the map.
Also, I ran some tests (benchs) on random data with different solutions (the one proposed by @timtofan, map, sort-iterate) and they are not much different.